tiledb/bioimg/visualization/helpers.py

```diff
@@ -0,0 +1,65 @@
+import json
+from dataclasses import dataclass
+from typing import List, Optional, Sequence
+
+import tiledb
+
+from ..converters.axes import Axes
+
+
+@dataclass(frozen=True)
+class ZoomLevelRecord:
+    zoom_level: int
+    image_level: int
+    downsample: int
+    width: int
+    height: int
+    namespace: str
+    array_uri: str
+    axes: str
+
+
+def calculate_metadata(
+    namespace: str, group_uri: str, config: Optional[tiledb.Config] = None
+) -> Sequence[ZoomLevelRecord]:
+    """
+    Calculates the required metadata for the Tile Image viewer.
+
+    :param namespace: the namespace of the group
+    :param group_uri: path to the group
+    :param config: the configuration for constructing the tileDB context
+    """
+    metadata: List[ZoomLevelRecord] = []
+
+    context = None if config is None else tiledb.Ctx(config)
+    group = tiledb.Group(group_uri, "r", ctx=context)
+    zoom_level = 0
+
+    for meta in json.loads(group.meta["levels"])[::-1]:
```

Comment:
1. `reversed(foo)` is more readable than `foo[::-1]` (and in theory more efficient for large sequences)
2. It's not (immediately) obvious why this has to be done in reversed level order, a comment would be nice.